### PR TITLE
improve kpt fn render error messages

### PR DESCRIPTION
**Pull request by natasha41575**
improve some error messages and add a test no-pipeline-in-subpackage

### CODE REVIEW

File: internal/fnruntime/container.go [hunk 1]
```diff
@@ -164,7 +164,7 @@ func (f *ContainerFn) prepareImage() error {
 	defer cancel()
 	cmd = exec.CommandContext(ctx, dockerBin, args...)
 	if _, err := cmd.CombinedOutput(); err != nil {
-		return fmt.Errorf("function image %q doesn't exist: %w", f.Image, err)
+		return fmt.Errorf("function image %q doesn't exist", f.Image)
```
[the remark this answers is not in the document]
Comment (Shell32-Natsu) [reply internal/fnruntime/container.go:167]: Yes, and add resolver in `internal/errors/resolver`

**Comment by natasha41575** [comment]
@droot please take a look!

File: internal/fnruntime/container.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (droot) [reply internal/fnruntime/container.go:167]: Adding specific errors such as `dockerError`, `pipelineValidatorError` in general is a good idea, so @Shell32-Natsu Go ahead with that.

As per this PR, this is a good improvement over current status, so I am going to approve this one.

File: e2e/testdata/fn-render/invalid-configmap-fnconfig/.expected/config.yaml [hunk 2]
```diff
@@ -13,4 +13,6 @@
 # limitations under the License.
 
 exitCode: 1
-stdErr: "line 14: cannot unmarshal !!seq into string"
\ No newline at end of file
+stdErr: |
+  Error: unable to parse "Kptfile": yaml: unmarshal errors:
+    line 14: cannot unmarshal !!seq into string
```
Comment (droot) [inline e2e/testdata/fn-render/invalid-configmap-fnconfig/.expected/config.yaml:18]: @mortent submitted another PR that improves the handling of kptfile parsing/missing related errors, so reconcile with that. https://github.com/GoogleContainerTools/kpt/pull/1977

**Review by droot — APPROVED** [review]

File: internal/fnruntime/container.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Shell32-Natsu) [reply internal/fnruntime/container.go:167]: Cool

**Comment by natasha41575** [comment]
@phanimarupaka @etefera I am assuming you will take care of the site check issue? Merged this with @droot's permission on that basis
